Docs Refresh (#213)
SleipnirGroup/Choreo: docs/usage/saving.md

# Saving

To access the sidebar, click the hamburger icon (top left of window).

![Sidebar image](../media/sidebar.png)

General Info:

> - Choreo’s file saving mechanism ties into a WPILib robot root directory, wherein a single `.chor` file lives in the project’s root folder.

Reviewer: IMO this would once again be better off as an info admonition than a single-item list.

> There are 5 options as well as a description below those pertaining to saving info.

Reviewer: ```suggestion
In the sidebar, there are five actions related to saving your project, along with your project's current save location.
```

## Open File

This opens the system’s file select dialog to select the robot’s `.chor` file. This should already be in your robot project root folder.

## Save File

This opens the system’s file save dialog to select where to save the robot’s `.chor` file. Choreo strongly recommends saving this file in your robot project root folder.

## New File

Creates a new file in memory, essentially clearing all trajectories. If you have unsaved changes, Choreo will ask before clearing them.

## Export Trajectory

> Exports the trajectory as a file individually as the path selected in the UI. This is not tied with the robot project structure, rather for you to get the trajectory file.

Reviewer: ```suggestion
Exports the trajectory as a file individually to the path you select in the UI. This is not tied with the robot project structure, so you can place the trajectory file anywhere you like.
```

## Save All Trajectories

This saves all the trajectories into the folder structure described below.

## Project Details

> If you have saved your choreo file, correctly, you should see the below:

Reviewer: ```suggestion
If you have saved your choreo file correctly, you should see the following:
```

![Project Info](../media/project_info.png)

> Project saved at shows the relative path where the `.chor` file lives in. You copy or open the full path in the default files app.

Reviewer: ```suggestion
Below "Project saved at", you can see the directory in which the saved `.chor` file lives. You can also copy this path or open it in your system's file explorer.
```

> Gradle project detected looks for a .gradle file as well as `/src/main/deploy/`, so it can put the trajectories in.

Reviewer: ```suggestion
If you see "Gradle project detected," then there is a `.gradle` file as well as a `src/main/deploy` directory in which to save trajectories.
```

> Trajectories saved in shows relative path where trajectories are saved in. [ChoreoLib](../choreolib/installation.md) looks inside the `{deployDirectory}/choreo` for a matching trajectory.

Reviewer: ```suggestion
Below "Trajectories saved in," you can see the path in which trajectories are saved.
Trajectories saved in shows relative path where trajectories are saved in. [ChoreoLib](../choreolib/installation.md) looks for trajectories in `{deployDirectory}/choreo`.
```

> <br><br><br>

Reviewer: This would again be better formatted as an admonition.

Also, why the random inconsistent spacing?

## New File

Creates a new file in memory, essentially clearing all trajectories. If you have unsaved changes, Choreo will ask before clearing them.

## Export Trajectory

Exports the trajectory as a file individually to the path you select in the UI. This is not tied with the robot project structure, so you can place the trajectory file anywhere you like.

## Save All Trajectories

This saves all the trajectories into the folder structure described below.

## Project Details

If you have saved your choreo file correctly, you should see the following:

![Project Info](../media/project_info.png)

Below "Project saved at", you can see the directory in which the saved `.chor` file lives. You can also copy this path or open it in your system's file explorer.

If you see "Gradle project detected," then there is a `.gradle` file as well as a `src/main/deploy` directory in which to save trajectories.

Below "Trajectories saved in," you can see the path in which trajectories are saved.
Trajectories saved in shows relative path where trajectories are saved in. [ChoreoLib](../choreolib/installation.md) looks for trajectories in `{deployDirectory}/choreo`.

> <br><br><br>

Reviewer 2: Remove these `<br>` tags.

If you get the below:

![Project not saved](../media/project_not_saved.png)

Then it means you have not saved the file yet.

> <br>

Reviewer 2: Including raw `<br>` tags like this is an antipattern.
